src/main/scala/akka/persistence/cassandra/CassandraPluginConfig.scala

```diff
@@ -79,55 +83,76 @@ class CassandraPluginConfig(config: Config) {
       connectionPoolConfig.getInt("pool-timeout-millis")
     )
 
-  val clusterBuilder: Cluster.Builder = {
-    val b = Cluster.builder
-      .addContactPointsWithPorts(contactPoints.asJava)
-      .withPoolingOptions(poolingOptions)
-      .withQueryOptions(new QueryOptions().setFetchSize(fetchSize))
-    protocolVersion match {
-      case None    => b
-      case Some(v) => b.withProtocolVersion(v)
+  def contactPoints(system: ActorSystem): Future[Seq[InetSocketAddress]] = {
+    val contactPointsProviderClassName = config.getString("contact-points-provider")
+    if (contactPointsProviderClassName.isEmpty) {
```

Comment:
Specifying the classname is the enabler, no need for two properties. It is similar with for example the snapshot plugin in Akka persistence, so yes, it is consistent with other things.